lms/djangoapps/teams/csv.py

```diff
@@ -276,14 +277,21 @@ def validate_user_assignment_to_team_and_teamset(self, row):
         user = row['user']
         for teamset_id in self.teamset_ids:
             team_name = row[teamset_id]
+            current_team_size = 0
             if not team_name:
                 continue
             try:
                 # checks for a team inside a specific team set. This way team names can be duplicated across
                 # teamsets
                 team = self.existing_course_teams[(team_name, teamset_id)]
+                current_team_size = team.users.count()
```

Comment:
Keeping this here means that we're making one query _per cell_ of the CSV.  Let's replace this line with these two lines:
```python
if (teamset_id, team_name) not in self.user_count_by_team:
    self.user_count_by_team[(teamset_id, team_name)] = team.users.count()
```
That should make it so that we make, at most, one of these queries per existing team.

Additionally, you can make `self.user_count_by_team` a `collections.Counter` object instead of a plain `dict`.  So if you fall into the `except` block here because the team doesn't exist, doing `self.user_count_by_team[(teamset_id, team_name)] += 1` below will still work, and result in that not-yet-existent team having a count of 1.